### ISSUE TITLE
plugin get blacklisted, libgstinterpipes-1.0.so is not a GStreamer plugin

### ISSUE THREAD

**Issue by braindevices** [RidgeRun/gst-interpipe#58]
```
$ gst-inspect-1.0 /tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so
Could not load plugin file: File "/tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so" is not a GStreamer plugin
$ gst-inspect-1.0 /tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so.1.0
Could not load plugin file: File "/tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so.1.0" is not a GStreamer plugin
```

I built it with gstreamer 1.17.90 or system gstreamer 1.16.2, both yield same problem.
here is configure result:
```
The Meson build system
Version: 0.55.1
Source dir: /tmp/gst-interpipe
Build dir: /tmp/gst-interpipe/build
Build type: native build
Using 'PKG_CONFIG_PATH' from environment with value: '/data/gst-build/1.17/lib/pkgconfig'
Using 'PKG_CONFIG_PATH' from environment with value: '/data/gst-build/1.17/lib/pkgconfig'
Project name: gst-interpipes
Project version: 1.1.2
Using 'CC' from environment with value: 'clang-10'
Using 'CC' from environment with value: 'clang-10'
C compiler for the host machine: clang-10 (clang 10.0.1 "Ubuntu clang version 10.0.1-++20200708122807+ef32c611aa2-1~exp1~20200707223407.61 ")
C linker for the host machine: clang-10 ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.6.3)
Using 'PKG_CONFIG_PATH' from environment with value: '/data/gst-build/1.17/lib/pkgconfig'
Run-time dependency gstreamer-app-1.0 found: YES 1.17.90
Using 'PKG_CONFIG_PATH' from environment with value: '/data/gst-build/1.17/lib/pkgconfig'
Run-time dependency gstreamer-check-1.0 found: YES 1.17.90
Compiler for C supports arguments -Wmissing-declarations: YES 
Compiler for C supports arguments -Wmissing-prototypes: YES 
Compiler for C supports arguments -Wredundant-decls: YES 
Compiler for C supports arguments -Wundef: YES 
Compiler for C supports arguments -Wwrite-strings: YES 
Compiler for C supports arguments -Wformat: YES 
Compiler for C supports arguments -Wformat-nonliteral: YES 
Compiler for C supports arguments -Wformat-security: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Waddress: YES 
Compiler for C supports arguments -Waggregate-return: YES 
Compiler for C supports arguments -Wno-multichar: YES 
Compiler for C supports arguments -Wdeclaration-after-statement: YES 
Compiler for C supports arguments -Wvla: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Has header "dlfcn.h" : YES 
Has header "inttypes.h" : YES 
Has header "memory.h" : YES 
Has header "poll.h" : YES 
Has header "stdint.h" : YES 
Has header "stdlib.h" : YES 
Has header "stdio_ext.h" : YES 
Has header "strings.h" : YES 
Has header "string.h" : YES 
Has header "sys/param.h" : YES 
Has header "sys/poll.h" : YES 
Has header "sys/prctl.h" : YES 
Has header "sys/socket.h" : YES 
Has header "sys/stat.h" : YES 
Has header "sys/times.h" : YES 
Has header "sys/time.h" : YES 
Has header "sys/types.h" : YES 
Has header "sys/utsname.h" : YES 
Has header "sys/wait.h" : YES 
Has header "ucontext.h" : YES 
Has header "unistd.h" : YES 
Has header "valgrind/valgrind.h" : NO 
Has header "sys/resource.h" : YES 
Using 'PKG_CONFIG_PATH' from environment with value: '/data/gst-build/1.17/lib/pkgconfig'
Run-time dependency gstreamer-1.0 found: YES 1.17.90
Configuring version.entities using configuration
Configuring config.h using configuration
Build targets in project: 15
```
ldd looks fine:
```
$ ldd build/gst/interpipe/libgstinterpipes-1.0.so
        linux-vdso.so.1 (0x00007ffdf47e3000)
        libgstapp-1.0.so.0 => /data/gst-build/1.17/lib/libgstapp-1.0.so.0 (0x00007fee168df000)
        libgstbase-1.0.so.0 => /data/gst-build/1.17/lib/libgstbase-1.0.so.0 (0x00007fee1685a000)
        libgstreamer-1.0.so.0 => /data/gst-build/1.17/lib/libgstreamer-1.0.so.0 (0x00007fee16703000)
        libgobject-2.0.so.0 => /lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fee16677000)
        libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fee1654e000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fee1635a000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fee16337000)
        libgmodule-2.0.so.0 => /lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007fee16331000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fee161e2000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fee161dc000)
        libunwind.so.8 => /lib/x86_64-linux-gnu/libunwind.so.8 (0x00007fee161bf000)
        libdw.so.1 => /lib/x86_64-linux-gnu/libdw.so.1 (0x00007fee1615e000)
        libffi.so.7 => /lib/x86_64-linux-gnu/libffi.so.7 (0x00007fee16152000)
        libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fee160df000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fee16905000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007fee160b6000)
        libelf.so.1 => /lib/x86_64-linux-gnu/libelf.so.1 (0x00007fee1609a000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fee1607c000)
        libbz2.so.1.0 => /lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007fee16069000)
```

I set the GST_PLUGIN_PATH
```
echo $GST_PLUGIN_PATH
/tmp/gst-interpipe/build/gst/interpipe/
```

but gst-inspect show it is blacklisted:
```
Blacklisted files:
  libgstvalidatessim.so
  libgstinterpipes-1.0.so

Total count: 2 blacklisted files

$ gst-inspect-1.0 libgstinterpipes-1.0.so
Plugin Details:
  Name                     libgstinterpipes-1.0.so
  Description              Plugin for blacklisted file
  Filename                 /tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so
  Version                  0.0.0
  License                  BLACKLIST
  Source module            BLACKLIST
  Binary package           BLACKLIST
  Origin URL               BLACKLIST


  0 features:

```

**Comment by michaelgruner** [comment]
Can you please provide the output of inspecting the library itself? 
gst-inspect-1.0 /tmp/gst-interpipe/build/gst/interpipe/libgstinterpipes-1.0.so
Michael

**Comment by stemcc** [comment]
I'm receiving the same error using Gstreamer 1.17.90, and when I attempt to inspect the library directly I'm receiving the same error that the plugin file `is not a GStreamer plugin`.

**Comment by braindevices** [comment]
> I'm receiving the same error using Gstreamer 1.17.90, and when I attempt to inspect the library directly I'm receiving the same error that the plugin file `is not a GStreamer plugin`.

Autogen works, the meson script has some bug.

**Comment by michaelgruner** [comment]
Thanks for the report. This is likely because the library does not match the plugin name. Definitely a bug in the newer versions of Gstreamer will fix and get back to you.

**Comment by jcaballeros** [comment]
Hi,

This should be fixed in release v1.1.3, currently in master. Closing, please feel free to reopen if the issue persists.
